.github/workflows/python.yml

```diff
@@ -1,5 +1,12 @@
 name: Python CI
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
We had this in the past on the config-service/cgw and the issue with this was that you'd not get CI feedback for pushed branches until you've opened a PR.

Not sure if you want this or not 😄

Reply:
I don't think that's the normal flow for this service as every branch usually is linked to a PR, only exceptions being master and develop (I forgot about develop and just added it). Thanks for the tip

Reply:
I get you @fmrsabino . Do you think that removing `pull_request` and only run CI on `push` would work? That way it will always run for PRs and for custom branches

Reply:
We had some issues with that in the past because if you remove `pull_request` then no action runs for forks – https://github.com/safe-global/safe-config-service/pull/404

Maybe there's a better approach to this.

For reference here are some other PRs that tried experimented with different setups:
- https://github.com/safe-global/safe-config-service/pull/5
- https://github.com/safe-global/safe-config-service/pull/25

Given this I think `pull_request` needs to stay. But the current solution that you propose also works (we just need to take into consideration that the CI would only run when opening a PR bus as long as we are fine with that then I think it should be good to go 👍 )

Reply:
Ok, then, makes sense, thanks for the info. I've just updated the PR